backend/src/apiserver/server/util.go

```diff
@@ -268,3 +273,39 @@ func CheckPipelineVersionReference(resourceManager *resource.ResourceManager, re
 
 	return &pipelineVersionId, nil
 }
+
+func GetUserIdentity(ctx context.Context) (string, error) {
+	if ctx == nil {
+		return "", nil
+	}
+	md, _ := metadata.FromIncomingContext(ctx)
+	// If the request header contains the user identity, requests are authorized
+	// based on the namespace field in the request.
+	if userIdentityHeader, ok := md[common.UserIdentityHeader]; ok {
+		if len(userIdentityHeader) != 1 {
+			return "", util.NewBadRequestError(errors.New("Request header error: user identity value is empty"), "Request header error: user identity value is empty")
+		}
+		userIdentityHeaderFields := strings.Split(userIdentityHeader[0], ":")
+		if len(userIdentityHeaderFields) != 2 {
+			return "", util.NewBadRequestError(errors.New("Request header error: user identity value is incorrectly formatted"), "Request header error: user identity value is incorrectly formatted")
+		}
+		return userIdentityHeaderFields[1], nil
+	}
+	return "", nil
+}
+
+func IsRequestAuthorized(resourceManager *resource.ResourceManager, userIdentity string, namespace string) (bool, error) {
```

Comment:
can we not wrap the function, instead just directly use resourceManager.IsRequestAuthorized where this function is called?

Reply:
Done.